src/aind_behavior_curriculum/curriculum.py

```diff
@@ -0,0 +1,1053 @@
+        """
+        template = Template(curr_dot_script)
+
+        # Add curriculum nodes
+        nodes = [
+            f'{node_id} [label="{node.name}"]'
+            for node_id, node in c.graph.nodes.items()
+        ]
+
+        # Add curriculum edges
+        edges = []
+        for start_id, edge_list in c.graph.graph.items():
+            for i, (edge, dest_id) in enumerate(edge_list):
+                # Use 1-indexing for labels
+                i = i + 1
+
+                # Edges must be StageTransition or PolicyTransition
+                edge_str = (
+                    f'{start_id} -> {dest_id} [label="({i}) '
+                    f'{edge.rule.__name__}", minlen=2]'
+                )
+                edges.append(edge_str)
+
+        curriculum_dot_script = template.render(
+            curr_name='"' + c.name + '"', nodes=nodes, edges=edges
+        )
+
+        return curriculum_dot_script
+
+    assert png_path.endswith(
+        ".png"
+    ), "Please add .png extension to end of png_path."
+    curr = validate_curriculum(curr)
+
+    dot_scripts = [make_curriculum_script(curr)]
+    last = []
+    for stage in curr.see_stages():
+        if stage.name == "GRADUATED":
+            last.append(make_stage_script(stage))
+            continue
+        dot_scripts.append(make_stage_script(stage))
+    dot_scripts = dot_scripts + last
+
+    # Finally concatenate these strings together in this order.
+    final_script = "\n".join(dot_scripts)
+
+    # Run graphviz export
+    gvpack_command = ["gvpack", "-u"]
+    dot_command = ["dot", "-Tpng", "-o", png_path]
+    gvpack_process = subprocess.Popen(
+        gvpack_command,
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.DEVNULL,
+    )
+    dot_process = subprocess.Popen(dot_command, stdin=subprocess.PIPE)
+
+    gvpack_output, _ = gvpack_process.communicate(input=final_script.encode())
+    dot_process.communicate(input=gvpack_output)
+
+
+def export_json(curr: Curriculum, json_path: str) -> None:
+    """
+    Export curriculum json to export path
+    """
+
+    assert json_path.endswith(
+        ".json"
+    ), "Please add .json extension to end of json_path."
+    curr = validate_curriculum(curr)
+
+    with open(json_path, "w") as f:
+        json_dict = curr.model_dump()
+        json_string = json.dumps(json_dict, indent=4)
+        f.write(json_string)
+
+
+def export_curriculum(curr: Curriculum, export_dir: str) -> None:
+    """
+    Export json and diagram into export dir.
+    """
+
+    export_json(curr, Path(export_dir) / "schema.json")
+    export_diagram(curr, Path(export_dir) / "diagram.png")
+
+
+# (This probably breaks)
+# To test after upload to bucket.
+def download_curriculum(
+    name: str, version: str, bucket="aind-behavior-curriculum-prod-o5171v"
+) -> Curriculum:
+    """
+    Reconstruct curriculum object from cloud json.
+    """
+
+    def read_json(bucket_name: str, json_key: Path | str) -> dict:
```

Comment:
I think I would rather do some sort of dependency injection where instead of passing the bucket_name to the method, I would pass the s3 object (or a wrapper class) instead.

Reply:
I think @dyf should wage in on this one. I have no idea what kind of API sci comp is providing researchers to interact with S3 buckets. If there is a library that is maintained elsewhere, I would like consider passing the object instead of creating it from scratch. If there is nothing in place, sure we can start with this and figure out things as we go... 
Regardless, I don't think this should prevent the PR from being merged, but def something we should consider.